pioinstaller/penv.py

```
@@ -16,18 +16,153 @@
+    return plat
+
+
+def _extract_uv_archive(archive_path, extract_dir):
+    """Extract uv archive to extract_dir and check for unsafe paths."""
+    if util.IS_WINDOWS:
+        with zipfile.ZipFile(archive_path) as zf:
+            base = os.path.abspath(extract_dir) + os.sep
+            for m in zf.infolist():
+                dest = os.path.abspath(os.path.join(extract_dir, m.filename))
+                if not dest.startswith(base):
+                    raise exception.PIOInstallerException("Unsafe path in archive")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)